cocotb/share/lib/simulator/simulatormodule.cpp

```diff
@@ -803,7 +793,6 @@ static PyObject *stop_simulator(PyObject *self, PyObject *args)
     COCOTB_UNUSED(self);
     COCOTB_UNUSED(args);
     gpi_sim_end();
```

Comment:
This now also causes the `FATAL: We are calling up again` error.
We aren't using it after this PR, should we remove it?

Reply:
This function is still useful. Users should be able to request shut down whenever, we can't always assume the simulator can determine when cocotb is done. For example, if you have an HDL clock, there will always be pending triggers and never shut down.